_posts/2019-11-25-z-index-is-confusing.md

```diff
@@ -0,0 +1,214 @@
+    }
+    .annoying p {
+        font-size: 2rem;
+        font-weight: 600;
+    }
+    .annoying__buttons {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+    }
+    .annoying__button {
+        -webkit-appearance: none;
+        appearance: none;
+        background-color: #84c2ff;
+        border-radius: 0.5rem;
+        color: #fff;
+        font-weight: 600;
+        padding: .5rem;
+        min-width: 10rem;
+        transition: all ease .2s;
+    }
+    .annoying__button:hover {
+        background-color: #fff;
+        color: #84c2ff;
+    }
+    .annoying__button--secondary {
+        background-color: #fff;
+        color: #84c2ff;
+    }
+    .annoying__button--secondary:hover {
+        background-color: #84c2ff;
+        color: #fff;
+    }
+    .annoying__button:last-of-type {
+        margin-left: 2rem;
+    }
+</style>
+
+For the longest time, I was guilty of thinking I knew how `z-index` really worked. I was mistaken. I never really took the time to research how it actually worked. I believe a lot of front end devs are guilty of the same thing when they're starting out. If we want an element to stack above another we use `z-index`. Simple, right? Give it a little `z-index`. Wait, why didn’t that work? It needs to be higher. Really high, in fact. Let’s give it `z-index: 999` yeah that ought to do it. Not necessarily. 
+
+Maybe, most importantly, we should first understand the default stacking order of elements even without a specified `z-index`. Simply put, elements are ordered from bottom to top. So the next sibling element in the dom will stack on top of the previous. Think about it like solitaire. When you deal out the cards, the first is laying on the table. As you find relevant cards, you place them next, and on top of the previous card.
+
+<figure>
+    <img src="https://cardgames.io/solitaire/images/solitaire-logo.png" style="max-width: 100%">
+    <figcaption>
+        In Solitaire, the cards stack with the most recent card on top of the pile.
+    </figcaption>
+</figure>
+
+Ok, that’s great that we got that out of the way, but let’s get to examples of those times when we have an element that needs to be stacked over another and `z-index: 9999` is not cutting it.
+
+I think the problem occurs most of the time when you have specifically positioned elements throughout the page. Maybe you have a fixed header or aside text block. Layouts like this can make for a complicated stacking order along the z-axis because defined position values, other than `position: static`, affect the stacking order.
+
+Without any specified `z-index`, the default stack occurs as previously mentioned. If any element in the stack receives a position value of absolute, relative, or fixed, that element will appear on top of the stack, or closest to the screen, so to speak.
+
+For example, say we have a site with a flyout in the header. To keep it simple nothing on this page has a specified `z-index` value. 
```

Comment:
comma nazi, `simple, `?

Reply:
https://app.grammarly.com/
they have free accounts ;)

Reply:
I used Grammarly. Maybe the free account is lazy?